sklearn/gaussian_process/gpr.py

```diff
@@ -236,6 +236,33 @@ def obj_func(theta, eval_gradient=True):
             # likelihood
             lml_values = list(map(itemgetter(1), optima))
             self.kernel_.theta = optima[np.argmin(lml_values)][0]
+            if np.isclose(self.kernel_.bounds,
+                          np.atleast_2d(self.kernel_.theta).T).any():
+                list_param = np.isclose(self.kernel_.bounds,
+                                        np.atleast_2d(self.kernel_.theta).T)
+                idx = 0
+                for hyp in self.kernel_.hyperparameters:
```

Comment:
This is very nested, and duplicated from gpc. Can we pull out either this warning logic, or the whole thing, into a helper function/method? It would be nice if this warning stuff was only a function of the mask of close values and the hyperparameters.

Reply:
> This is very nested, and duplicated from gpc. Can we pull out either this warning logic, or the whole thing, into a helper function/method? It would be nice if this warning stuff was only a function of the mask of close values and the hyperparameters.

Where would you see such a helper function ? 
And I know it's very nested but I didn't see any clearer way of doing it that matched all different scenarios.

Reply:
Where? In gaussian_process.kernels, or a new gaussian_process.base.

I don't think this warning is a function of many variables, so you can happily pull it into a separate, encapsulated scope, avoiding nesting.